### PR TITLE
Adding Signup page

### LINKED ISSUE #26: Create the signup page
The scope of this ticket is to  create a signup page in ionic project.

**Acceptance criteria**
The designs should have the following:

- [ ]  A centred card component with the following input fields 

- Name
- Email
- Password
- [ ] A submit buttton with primary color
- [ ] Make sure that the page is designed mobile first

### PR DESCRIPTION
**Pull request by sirajulmcb**
* Signup form created using Ionic and Formik
* Added dummy Home page
* Added store and api from praas-app
* Added validation for signup form

Fixes #26

### CODE REVIEW

**Comment by sirajulmcb** [comment]
@vakareeralil Most of the files are copied directly from `praas-app`, like `store, `api`, etc. Since I dint find any benefit doing them again.

File: praas-app-ionic/package.json [hunk 1]
```diff
@@ -12,15 +12,19 @@
     "@types/node": "^12.12.24",
     "@types/react": "^16.9.17",
     "@types/react-dom": "^16.9.4",
-    "@types/react-router": "^5.1.4",
-    "@types/react-router-dom": "^5.1.3",
+    "axios": "^0.19.2",
```
[the remark this answers is not in the document]
Comment (sirajulmcb) [reply praas-app-ionic/package.json:15]: Ah. I added it initially. but decided to use existing api login from praas-app

File: praas-app-ionic/src/pages/Signup/index.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    if (user.login.loggedIn) history.replace('/');
+  }, [user, history]);
+
+  console.log(formik.errors);
+
+  return (
+    <IonPage className="signup-page">
+      <Header />
+      <IonContent>
+        <form noValidate onSubmit={formik.handleSubmit}>
```
[the remark this answers is not in the document]
Comment (sirajulmcb) [reply praas-app-ionic/src/pages/Signup/index.tsx:62]: I usually add this to ignore default html form validation when doing a custom validation
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form

But it seems to work without it. Removing it for now

File: praas-app-ionic/src/pages/Signup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  const formik = useFormik({
+    initialValues: { firstName: '', email: '', password: '' },
+    validationSchema: signupSchema,
+    onSubmit: (values, actions) => {
+      registerUser({ user: values }, actions);
+    }
+  });
+
+  useEffect(() => {
+    if (user.login.loggedIn) history.replace('/');
```
[the remark this answers is not in the document]
Comment (sirajulmcb) [reply praas-app-ionic/src/pages/Signup/index.tsx:53]: Yes I prefer always block option. prettier was automatically removing the block styles. I also prefer a trailing comma for objects and multi line.

File: praas-app-ionic/src/pages/Signup/index.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        <form noValidate onSubmit={formik.handleSubmit}>
+          <IonGrid>
+            <IonRow className="ion-justify-content-center">
+              <IonCol sizeXs="12" sizeSm="4" className="text-align-center">
+                <IonItem>
+                  <IonLabel position="floating">Name</IonLabel>
+                  <IonInput
+                    type="text"
+                    name="firstName"
+                    onIonChange={event => {
```
[the remark this answers is not in the document]
Comment (sirajulmcb) [reply praas-app-ionic/src/pages/Signup/index.tsx:71]: Yes saw this sample. I'm working on this approach.

File: praas-app-ionic/package.json [hunk 5]
```diff
@@ -12,15 +12,19 @@
...
+    "node-sass": "^4.13.1",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
+    "react-redux": "^7.2.0",
     "react-router": "^5.1.2",
     "react-router-dom": "^5.1.2",
     "react-scripts": "3.4.0",
-    "typescript": "3.7.4"
+    "redux-thunk": "^2.3.0",
+    "typescript": "3.7.4",
```
[the remark this answers is not in the document]
Comment (sirajulmcb) [reply praas-app-ionic/package.json:26]: Yes those are the files copied from  praas-app. Yes we need to move them to ts and add proper types. 
If we need to convert them now. I would rename them to .ts and we could add the types slowly when working more on it? What is your opinion?

**Comment by vakareeralil** [comment]
@sirajulmcb Also did you see to react-hooks-form? Going ahead with that could be a better solution to avoid clutter in the UI. Unlike formik, it works well with ionic react is what I read. Let me know your thoughts.

File: praas-app-ionic/src/pages/Signup/index.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+import Header from '../../components/Header';
+import { registerUser } from '../../store/user/registration';
+import './style.scss';
+
+type Props = {
+  user: any;
+  registerUser: (data: any, formikActions: any) => void;
+};
+
+const signupSchema = Yup.object().shape({
```
Comment (vakareeralil) [inline praas-app-ionic/src/pages/Signup/index.tsx:26]: Can we also aggregate the schemas?

**Comment by vakareeralil** [comment]
> For development purpose npm install is the better candidate, it would not regenerate package-lock.json if the package is present, or unless you update package.json or install a diff package version.
> npm ci always deletes node_modules and installs the packages again, which is bit painful in my opinion 😛
> But it suits well in CI environments.
> https://stackoverflow.com/questions/52499617/what-is-the-difference-between-npm-install-and-npm-ci

https://blog.quigley.codes/everything-you-wanted-to-know-about-package-lock-json/?dr=https://medium.com/p/b81911aa8ab8 @sirajulmcb this is a good read

File: praas-app-ionic/src/pages/Signup/index.tsx [hunk 7]
```diff
@@ -40,62 +41,43 @@ const Signup: React.FC<Props & RouteComponentProps> = ({
...
+  const { control, register, handleSubmit, errors, formState } = useForm({
+    defaultValues: { firstName: '', email: '', password: '' },
     validationSchema: signupSchema,
-    onSubmit: (values, actions) => {
-      registerUser({ user: values }, actions);
-    },
   });
 
+  const onSubmit = (values: any) => {
+    debugger;
```
Comment (vakareeralil) [inline praas-app-ionic/src/pages/Signup/index.tsx:50]: hmm.. this is gonna create some trouble 🗡️

**Pull request merged**